src/xenia/vfs/devices/stfs_container_file.cc

```diff
@@ -52,12 +52,15 @@ X_STATUS StfsContainerFile::ReadSync(void* buffer, size_t buffer_length,
     size_t read_length =
         std::min(record.length - read_offset, remaining_length);
 
-    auto& file = entry_->files()->at(record.file);
-    xe::filesystem::Seek(file, record.offset + read_offset, SEEK_SET);
-    auto num_read = fread(p, 1, read_length, file);
+    auto global_lock = global_critical_region_.Acquire();
```

Comment:
Is that a race between guess threads accessing the _same_ guest file handle, or _different_ guest handles implemented through one host file handle? If that's the latter, we should be opening the host container file multiple times, for each guest handle, or if possible, use a memory mapping. Using the global critical region here will create a lot of contention that may block other threads for milliseconds — the comment in `xenia/base/mutex.h` explicitly says that no I/O must be done there.

Reply:
Ohh my bad for not reading comments in mutex.h. 
It uses multiple handles, so rewrites then to allow memory mapping probably

Reply:
That's strange, I thought it _was_ already using memory mapping for STFS mapping? Maybe I'm forgetting when it got changed. I know I wanted the capability to do both, since we need the ability to load STFS packages nested within another STFS package.